design/accepted/gc_plus/gc-plus-milestone2-execution-plan.md

```diff
@@ -0,0 +1,50 @@
+# Uncommitted Garbage Collection - Milestone 2 Execution Plan
+
+Uncommitted Garbage Collection [Proposal](https://github.com/treeverse/lakeFS/blob/master/design/accepted/gc_plus/uncommitted-gc.md)
+
+## Milestone 1
+The first beta version that was released included:
+1. Implementation of the clean run flow for old and new repository structures (without optimizations)
+2. Mark & Sweep
+3. Integration tests
+4. Backup & Restore
+
+## Milestone 2
+
+### Goals
+1. Removing the limitation of a read-only lakeFS during the job run
+2. Performance improvements - better parallelization of the storage namespace traversal
+3. Implementing Optimized Run
```

Comment:
2 and 3 are performance and optimizations - the first item is self explained, I didn't understand the second one.
Do you mean incremental uncommitted gc?

Reply:
@nopcoder Optimized run means incremental
@eden-ohana I think we should focus this milestone on optimization of the full run and adding more testing (unit / integration) - and create a separate milestone for the incremental/optimized run